test/src/test/java/org/corfudb/integration/MergeTestListener.java

```diff
@@ -0,0 +1,204 @@
+     * existingKeysInMergedTable from addedKeys and deletedKeys.
+     */
+    @Override
+    protected void onSnapshotSyncComplete() {
+        try (TxnContext tx = corfuStoreSink.txn(NAMESPACE)){
+            existingKeysInMergedTable.addAll(addedKeys.keySet());
+            addedKeys.forEach((addedKey, e) -> {
+                tx.putRecord(tx.getTable(MERGED_TABLE_NAME), addedKey,
+                        SampleSchema.SampleMergedTable.newBuilder().setPayload(e.getPayload().toString().replaceAll("[^0-9]", "")).build(), e.getMetadata());
+            });
+
+            existingKeysInMergedTable.removeAll(deletedKeys);
+            deletedKeys.forEach(deletedKey -> {
+                tx.delete(tx.getTable(MERGED_TABLE_NAME), deletedKey);
+            });
+
+            tx.commit();
+        }
+
+        log.info("Snapshot sync complete");
+    }
+
+    /**
+     * Write local tables to merged table, process updates and add the keys to addedKeys
+     * and deletedKeys for replicated table.
+     *
+     * @param results Entries received in a single transaction as part of a snapshot sync
+     */
+    @Override
+    protected void processUpdatesInSnapshotSync(CorfuStreamEntries results) {
+        log.info("Processing updates in snapshot sync");
+        results.getEntries().forEach((schema, entries) -> {
+            entries.forEach(e -> {
+                Sample.StringKey key = (Sample.StringKey) e.getKey();
+                if (schema.getTableName().contains("Local_Table00")) {
+                    if (!(e.getOperation() == CorfuStreamEntry.OperationType.CLEAR) &&
+                            !(existingKeysInMergedTable.contains(key))) {
+                        try (TxnContext tx = corfuStoreSink.txn(namespace)){
+                            if (e.getOperation() == CorfuStreamEntry.OperationType.UPDATE) {
+                                tx.putRecord(tx.getTable(MERGED_TABLE_NAME), key,
+                                                SampleSchema.SampleMergedTable.newBuilder().setPayload(e.getPayload().toString().replaceAll("[^0-9]", "")).build(),
+                                         e.getMetadata());
+                                existingKeysInMergedTable.add(key);
+                            }
+                            else {
+                                tx.delete(tx.getTable(MERGED_TABLE_NAME), key);
+                            }
+                            tx.commit();
+                        }
+                    }
+                }
+                else {
+                    if ((e.getOperation() == CorfuStreamEntry.OperationType.UPDATE) &&
+                            !(existingKeysInMergedTable.contains(key))) {
+                        updates.add(e);
+                        addedKeys.put(key, e);
+                    }
+                    else if ((e.getOperation() == CorfuStreamEntry.OperationType.DELETE) &&
+                            !(existingKeysInMergedTable.contains(key))) {
```

Comment:
This condition itself is invalid right? If the key doesn't even exist in `existingKeysInMergedTable`, it shouldn't expect to be DELETED

Reply:
You are right. Just removed the `!` for all `DELETE` operation's checks.